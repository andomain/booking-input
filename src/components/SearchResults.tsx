import React, { FC } from 'react';
import { SearchResponse } from '../types';

import SearchResult from './SearchResult';

interface SearchResultsProps {
  results: SearchResponse[] | null;
  onSelect: (name: string, data: SearchResponse) => void;
  reset: () => void;
}

/**
 * <Search Results />
 * Wrapper component for <searchResult /> components
 */
const SearchResults: FC<SearchResultsProps> = ({
  results,
  onSelect,
  reset,
}) => {
  return (
    <div className="SearchResults">
      <ul>
        {results !== null &&
          results.length > 0 &&
          results.map(result => (
            <SearchResult
              key={result.placeKey}
              data={result}
              onSelect={onSelect}
            />
          ))}
        {results !== null && results.length === 0 && (
          <li className="SearchResult SearchResult--none" onClick={reset}>
            No results found
          </li>
        )}
      </ul>
    </div>
  );
};

export default SearchResults;
