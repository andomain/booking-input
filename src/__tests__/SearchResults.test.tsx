import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import SearchResults from '../components/SearchResults';

describe('<SearchResults />', () => {
  let wrapper: ReactWrapper;

  const mockSelect = jest.fn();
  const mockReset = jest.fn();

  beforeEach(() => {
    wrapper = mount(
      <SearchResults onSelect={mockSelect} reset={mockReset} results={[]} />,
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
