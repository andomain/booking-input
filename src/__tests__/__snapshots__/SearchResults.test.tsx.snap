// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> should render correctly 1`] = `
<SearchResults
  onSelect={[MockFunction]}
  reset={[MockFunction]}
  results={Array []}
>
  <div
    className="SearchResults"
  >
    <ul>
      <li
        className="SearchResult SearchResult--none"
        onClick={[MockFunction]}
      >
        No results found
      </li>
    </ul>
  </div>
</SearchResults>
`;
